pkg/apis/core/validation/shoot.go

```diff
@@ -2218,3 +2221,11 @@ func isShootReadyForRotationStart(lastOperation *core.LastOperation) bool {
 	}
 	return lastOperation.Type == core.LastOperationTypeReconcile
 }
+
+func isShootInHibernation(shoot *core.Shoot) bool {
+	if shoot.Spec.Hibernation != nil {
+		return *shoot.Spec.Hibernation.Enabled
+	}
+
+	return shoot.Status.IsHibernated
+}
```

Comment:
Q- 
Waking up of the cluster will not have any issues in scaling right? When `shoot.Spec.Hibernation.Enabled` is false and `shoot.Status.IsHibernated` is true?

Reply:
You'd never know. From an etcd pov, it would take some time for the etcd to start up. A cluster scaling up before the etcd is up could lead to the same error

Reply:
When the cluster is waking up `shoot.Spec.Hibernation` will still be non-nil but `shoot.Spec.Hibernation.Enabled` will be false. Isn't it? We never return `shoot.Status.IsHibernated` since  `shoot.Spec.Hibernation` is always non-nil.
Maybe we should check the status always?

Reply:
Yes, @acumino pointed that out earlier today and I just made a change correcting that
Can you take a look at the latest commit? it should solve your issue

Reply:
I think https://github.com/gardener/gardener/pull/7894/commits/1057321f38d07b440d88958940dcbdf89cd49ee2 fixes my concern